src/credentials/wizards/connections.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import { getIcon, codicon } from '../../shared/icons'
+import { telemetry } from '../../shared/telemetry/telemetry'
+import { createRefreshButton, createExitButton } from '../../shared/ui/buttons'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { Auth, Connection, isIamConnection, reauthCommand } from '../auth'
+import { getDependentAuths } from '../secondaryAuth'
+
+export const getConnectionIcon = (conn: Connection) =>
+    conn.type === 'sso' ? getIcon('vscode-account') : getIcon('vscode-key')
+
+export function createConnectionPrompter(auth: Auth, type?: 'iam' | 'sso') {
```

Comment:
Had to split this out due to potential circular dependency issues. `secondaryAuth.ts` could possibly be consolidated into `auth.ts` to simplify things.

Reply:
Nevermind. Splitting up the file isn't a huge amount of work but it does require moving around a lot of stuff which I'd rather do in a separate PR.